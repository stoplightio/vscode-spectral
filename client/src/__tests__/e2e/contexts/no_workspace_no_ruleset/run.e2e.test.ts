/* eslint-disable require-jsdoc */

import { expect } from 'chai';
import * as chaiJestSnapshot from 'chai-jest-snapshot';
import * as vscode from 'vscode';
import * as path from 'path';

import { openFile, activate } from '../../helper';

suiteSetup(async () => {
  chaiJestSnapshot.resetSnapshotRegistry();
  await activate();
});

setup(function() {
  // eslint-disable-next-line no-invalid-this
  chaiJestSnapshot.configureUsingMochaContext(this);
});

suite('No workspace, no ruleset', () => {
  suite('No diagnostics for empty files', () => {
    ['empty.yaml', 'empty.json'].forEach((fixture) => {
      test(`${fixture}`, async () => {
        const diags = await lint(['empty', fixture]);
        expect(diags).to.matchSnapshot();
      });
    });
  });

  suite('No diagnostics for valid files', () => {
    ['simple.yaml', 'simple.json'].forEach((fixture) => {
      test(`${fixture}`, async () => {
        const diags = await lint(['valid', fixture]);
        expect(diags).to.matchSnapshot();
      });
    });
  });

  suite('Invalid files trigger generation of diagnostics', () => {
    ['simple.yaml', 'simple.json'].forEach((fixture) => {
      test(`${fixture}`, async () => {
        const diags = await lint(['invalid', fixture]);

        expect(diags).to.matchSnapshot();
      });
    });
  });

  const pathToFixtures = '../../fixtures';

  const lint = async (pathSegments: string[]) => {
    const docPath = path.resolve(__dirname, pathToFixtures, ...pathSegments);

    const docUri = vscode.Uri.file(docPath);
    await openFile(docUri);

    return vscode.languages.getDiagnostics(docUri);
  };
});
