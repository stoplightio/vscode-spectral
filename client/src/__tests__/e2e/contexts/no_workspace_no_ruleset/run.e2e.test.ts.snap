// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No workspace, no ruleset Invalid files trigger generation of diagnostics simple.json 1`] = `
Array [
  Object {
    "code": "oas3-api-servers",
    "message": "OpenAPI \\"servers\\" must be present and non-empty array.",
    "range": Array [
      Object {
        "character": 0,
        "line": 0,
      },
      Object {
        "character": 12,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "oas3-schema",
    "message": "Object must have required property \\"paths\\".",
    "range": Array [
      Object {
        "character": 0,
        "line": 0,
      },
      Object {
        "character": 12,
        "line": 2,
      },
    ],
    "severity": "Error",
    "source": "spectral",
  },
  Object {
    "code": "info-contact",
    "message": "Info object must have \\"contact\\" object.",
    "range": Array [
      Object {
        "character": 10,
        "line": 2,
      },
      Object {
        "character": 12,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "info-description",
    "message": "Info \\"description\\" must be present and non-empty string.",
    "range": Array [
      Object {
        "character": 10,
        "line": 2,
      },
      Object {
        "character": 12,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "oas3-schema",
    "message": "\\"info\\" property type must be object.",
    "range": Array [
      Object {
        "character": 10,
        "line": 2,
      },
      Object {
        "character": 12,
        "line": 2,
      },
    ],
    "severity": "Error",
    "source": "spectral",
  },
]
`;

exports[`No workspace, no ruleset Invalid files trigger generation of diagnostics simple.yaml 1`] = `
Array [
  Object {
    "code": "oas3-api-servers",
    "message": "OpenAPI \\"servers\\" must be present and non-empty array.",
    "range": Array [
      Object {
        "character": 0,
        "line": 0,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "oas3-schema",
    "message": "Object must have required property \\"paths\\".",
    "range": Array [
      Object {
        "character": 0,
        "line": 0,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Error",
    "source": "spectral",
  },
  Object {
    "code": "info-contact",
    "message": "Info object must have \\"contact\\" object.",
    "range": Array [
      Object {
        "character": 6,
        "line": 2,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "info-description",
    "message": "Info \\"description\\" must be present and non-empty string.",
    "range": Array [
      Object {
        "character": 6,
        "line": 2,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Warning",
    "source": "spectral",
  },
  Object {
    "code": "oas3-schema",
    "message": "\\"info\\" property type must be object.",
    "range": Array [
      Object {
        "character": 6,
        "line": 2,
      },
      Object {
        "character": 8,
        "line": 2,
      },
    ],
    "severity": "Error",
    "source": "spectral",
  },
]
`;

exports[`No workspace, no ruleset No diagnostics for empty files empty.json 1`] = `Array []`;

exports[`No workspace, no ruleset No diagnostics for empty files empty.yaml 1`] = `Array []`;

exports[`No workspace, no ruleset No diagnostics for valid files simple.json 1`] = `Array []`;

exports[`No workspace, no ruleset No diagnostics for valid files simple.yaml 1`] = `Array []`;
